### PR TITLE
feat(payment): PI-1586 Refactor the existing `zip` payment strategy in Checkout SDK

### PR DESCRIPTION
**Pull request by bc-ania**
## What?
Refactor the existing `zip` payment strategy in Checkout SDK

## Testing / Proof
unit tests
Screenshot 2024-04-26 at 13 25 02

https://github.com/bigcommerce/checkout-sdk-js/assets/130754705/7733833a-18af-4f50-b5f6-118fac2a3e4c

@bigcommerce/team-checkout @bigcommerce/team-payments

### CODE REVIEW

File: packages/payment-integration-api/src/payment/storefront-payment-request-sender.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { RequestSender } from '@bigcommerce/request-sender';
+
+import { ContentType, INTERNAL_USE_ONLY, SDK_VERSION_HEADERS } from '../common/http-request';
+
+export default class StorefrontPaymentRequestSender {
```
Comment (animesh1987) [inline packages/payment-integration-api/src/payment/storefront-payment-request-sender.ts:5]: Is this implemented by other providers as well?

Comment (bc-ania) [reply packages/payment-integration-api/src/payment/storefront-payment-request-sender.ts:5]: Yes, quadpay and opy use it too

Comment (animesh1987) [reply packages/payment-integration-api/src/payment/storefront-payment-request-sender.ts:5]: Thanks, I feel this is something we can just do directly in the payment strategy. Do we need to expose a shared method here?

**Review by bc-peng — APPROVED** [review]
LGTM.👏

**Review by animesh1987 — APPROVED** [review]

**Review by bc-wpietrzak — APPROVED** [review]
LGTM

**Pull request merged**